FaunaDB.Client/Types/RefParser.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace FaunaDB.Types
+{
+    internal static class RefParser
+    {
+        public static RefV Parse(Value value)
+        {
+            return value.GetOption(Field.At("@ref")).Match(
+                Some: Parse,
+                None: () =>
+                {
+                    var id = value.GetOption(Field.At("id"));
+                    var cls = value.GetOption(Field.At("class"));
+                    var db = value.GetOption(Field.At("database"));
+
+                    return Mk(id, cls, db);
+                }
+            );
+        }
+
+        static RefV Mk(IOption<Value> id, IOption<Value> cls, IOption<Value> db)
+        {
+            var idE = ParseId(id);
+            var classE = Cast<ClassV>(cls);
+            var databaseE = Cast<DatabaseV>(db);
+
+            if (classE == null && databaseE == null)
+                return BuiltIn.FromName(idE);
+
+            if (classE == null)
+                return new ClassV(id: idE, database: databaseE);
+
+            if (BuiltIn.DATABASES == classE)
+                return new DatabaseV(id: idE, database: databaseE);
+
+            if (BuiltIn.CLASSES == classE)
+                return new ClassV(id: idE, database: databaseE);
+
+            if (BuiltIn.INDEXES == classE)
+                return new IndexV(id: idE, database: databaseE);
+
+            if (BuiltIn.FUNCTIONS == classE)
+                return new FunctionV(id: idE, database: databaseE);
+
+            if (BuiltIn.KEYS == classE)
+                return new KeyV(id: idE, database: databaseE);
+
+            if (databaseE == null)
+                return new RefV(id: idE, @class: classE);
+
+            return null;
+        }
+
+        static string ParseId(IOption<Value> id)
+        {
+            return id.FlatMap(value =>
+            {
+                return value.To(Codec.STRING).Match(
```

Comment:
I was digging into this. Core will always return ref's id as a `String`. We can remove this check.